File: ArtBookFragment/app/src/main/java/com/tolgaalperkus/artbookfragment/MainActivity.java
package com.tolgaalperkus.artbookfragment;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteStatement;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;

import java.io.ByteArrayOutputStream;

public class MainActivity extends AppCompatActivity {

    public static SQLiteDatabase database;
    public static int pathInfo =0;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        database = this.openOrCreateDatabase("Arts",MODE_PRIVATE,null);

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.options_menu,menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if(item.getItemId() == R.id.options_menu_add_art_item){
            if(pathInfo == 0){
                ListFragmentDirections.ActionListFragmentToİtemFragment action = ListFragmentDirections.actionListFragmentToİtemFragment();
                action.setInfo(0);
                Navigation.findNavController(MainActivity.this,R.id.fragment2).navigate(action);
            }else{
                NavDirections action = ItemFragmentDirections.actionİtemFragmentSelf();
                Navigation.findNavController(MainActivity.this,R.id.fragment2).navigate(action);
            }

        }
        return super.onOptionsItemSelected(item);
    }
}

File: ViewBindingKotlin/app/src/main/java/com/tolgaalperkus/viewbindingkotlin/MainActivity.kt
package com.tolgaalperkus.viewbindingkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import com.tolgaalperkus.viewbindingkotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    //private lateinit var textView: TextView
    private lateinit var binding : ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        //textView = findViewById<TextView>(R.id.textView)
    }

    fun ismiDegistir(view : View){
        println("degistir tiklandi")

       // textView.text = "Merhaba kotlin"
        binding.textView.text = "Merhaba Kotlin"
    }
}